### PR TITLE
Allow user to specify boost version

### PR DESCRIPTION
**Pull request by meshula**
### Description of Change(s) 

I'm using boost 1.65.1, and like the convenience of using the standard build script. This modification doesn't change the default behavior, but if boost version is specified on the command line, USD will build with the specified version.

Also handles toolset version for VS2015 and VS2017, because boost 1.65.1 insists on toolset 141 for VS2017.

OpenImageIO 1.7.17 is the minimum compatible with boost 1.65.1

### Fixes Issue(s) 

build script is hard coded to boost 1.61 and using the 140 toolset with VS2015 and VS2017

### CODE REVIEW

**Comment by jtran56** [comment]
Filed as internal issue #154998.

File: build_scripts/build_usd.py [hunk 1]
```diff
@@ -1143,6 +1155,19 @@ def ForceBuildDependency(self, dep):
...
     os.environ['PYTHONPATH'] = os.pathsep.join(paths)
 
+# The default installation of boost on Windows puts headers in a versioned 
+# subdirectory, which we have to account for here. In theory, specifying 
+# "layout=system" would make the Windows install match Linux/MacOS, but that 
+# causes problems for other dependencies that look for boost.
+if Windows():
+    # .0 versions of boost don't end in _0.
+    boost_version = "_".join(context.boost.split("."))
+    if boost_version[-2:] == ".0":
```
Comment (meshula) [inline build_scripts/build_usd.py]: Typo - this should be == "_0":

**Comment by sunyab** [comment]
We'd prefer to keep build_usd.py to a curated set of dependencies for easier testing and validation, especially given the relationships between dependencies like the one between boost and OpenImageIO identified in this PR. This sort of customization (even at an "at-your-own-risk" level) opens us up to more support load; if users want to use different versions they can edit the build script locally, which should make it clear they're in unguarded waters. :)

So, I'm going to close this PR out. Thanks!

**Comment by meshula** [comment]
Makes sense to me, the curated build environment today works for me very well in all the configurations I'm dealing with!

**Pull request closed without merging**